### PR TITLE
Improved cal_tc.py for decomposing heat conductivity

### PR DESCRIPTION
**Pull request by shigure1223**
I have changed the cal_tc.py so that the conductivity will be decomposed.
The data will be output with no problems, whenever decomp is set "no" in the config file.

### CODE REVIEW

**Comment by flat35hd99** [comment]
Thank you for your PR! I'll read your changes tonight(19:00-)

File: curp/script/cal_tc.py [hunk 1]
```diff
@@ -37,6 +37,14 @@ def get_dt(flux_fn):
 
     return d_t
 
+def get_decomp(flux_fn):
+
+    ncfile = netcdf.Dataset(flux_fn, mode='r')
+    len_decomps = len(ncfile.dimensions['ncomponent'])
+    ncfile.close()
+
+
```
Comment (flat35hd99) [inline curp/script/cal_tc.py:46]: Remove this empty line. You don't need two lines to separate code blocks.

**Review by flat35hd99 — CHANGES_REQUESTED** [review]

**Comment by flat35hd99** [comment]
I would like you to change single points that I mentioned above.

Additionaly, I request you to do:

- `The data will be output with no problems, whenever decomp is set "no" in the config file.` Share the config file and options that you specified(both of flux and conductivity calculations). I cannot understand your test situation.
- If you have not checked the energy flux & conductivity calculations, please perform that test.(this is called "regression test")

**Comment by shigure1223** [comment]
Thank you for the comments.
I deleted the empty line.

I will share the config file and options that I specified(both of flux and conductivity calculations) below.

1. config file
```python
[input]
format = amber
first_last_interval = 1 999999 1
#group_file = atomgroup.dat

[input_amber]
target = trajectory
topology_file = system.dry.prmtop
coordinate_format = netcdf
coordinate_file = md.crd.nc
velocity_format = netcdf
velocity_file = md.vel.nc

[curp]
potential = amber12SB
method = heat-flux

group_method = residue
#group_method = file
flux_grain = group
group_pair_file = gpair.dat

remove_trans =  yes
remove_rotate = yes

log_frequency = 1000

[output]
filename = flux.nc
format = netcdf
decomp = no

output_energy = no
```

2. job file
```python
curp compute flux.cfg > $output/flux.log

curp cal-tc \
    --frame-range 1 60000 1 --average-shift 2 \
    --dt 0.001 \
    -a $output/acf.nc \
    -o $output/conductivity.dat \
    $output/flux.nc > $output/conductivity.log
```

**Comment by shigure1223** [comment]
I finished the regression test of energy & heat conductivity calculations.

1. For decomp=no, both energy & heat conductivity calculations had no changes in the result with curp1.3.1
2. For decomp=yes heat conductivity calculations, the total conductivity in "conductivity.dat" was the same as the decomp=no heat conductivity calculations
3. For decomp=no energy calculations, the result had no changes but there were one problem that some conductivity was printed 'nan'  (because we cannot calculate the conductivity of bonded interaction term when residues are apart).This is a natural result but to output the data into a manageable form, I changed like below

https://github.com/yamatolab/current-calculations-for-proteins/pull/34/commits/8a0b2af3180d631b4b6cfd13979ce987123a7dde

If there are other things I have to check , please let me know.
Thank  you.

**Comment by shigure1223** [comment]
Yes correct!!!

**Review by flat35hd99 — APPROVED** [review]
LGTM and the regression test was performed by @shigure1223 . It is good to merge.

**Comment by flat35hd99** [comment]
@yamato97 This PR provides decompose conductivity into conductivity of each force item. @shigure1223 also check this change make no side effect.

Please check and if this is okay, approve this PR.

**Comment by yamato97** [comment]
LGTM too.

**Comment by yamato97** [comment]
LGTM too

**Review by yamato97 — APPROVED** [review]

**Pull request merged**